cmd/elasticsearch-operator/main.go

```diff
@@ -13,13 +17,77 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+const (
+	logFormatLogfmt = "logfmt"
+	logFormatJSON   = "json"
+	logLevelPanic   = "panic"
+	logLevelFatal   = "fatal"
+	logLevelError   = "error"
+	logLevelWarn    = "warn"
+	logLevelInfo    = "info"
+	logLevelDebug   = "debug"
+)
+
+var (
+	logLevel           string
+	logFormat          string
+	availableLogLevels = []string{
+		logLevelPanic,
+		logLevelFatal,
+		logLevelError,
+		logLevelWarn,
+		logLevelInfo,
+		logLevelDebug,
+	}
+	availableLogFormats = []string{
+		logFormatLogfmt,
+		logFormatJSON,
+	}
+)
+
 func printVersion() {
 	logrus.Infof("Go Version: %s", runtime.Version())
 	logrus.Infof("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH)
 	logrus.Infof("operator-sdk Version: %v", sdkVersion.Version)
 }
 
-func main() {
+func init() {
+	flagset := flag.CommandLine
+	flagset.StringVar(&logLevel, "log-level", logLevelInfo, fmt.Sprintf("Log level to use. Possible values: %s", strings.Join(availableLogLevels, ", ")))
```

Comment:
Is this consistent with the oc binary?  If not, please make it so.

Reply:
No, oc binary uses integer numbering for expressing severity
https://blog.openshift.com/oc-command-newbies/

However it is consistent with other operators that allow users to change log lever/format. (To my surprise almost no operators have this...)
https://github.com/coreos/prometheus-operator/blob/master/cmd/operator/main.go#L132-L133